main/ahb_for_sles.go

```diff
@@ -97,13 +98,49 @@ func parseCfg(filename string) (map[string]map[string]string, error) {
 	return ini, nil
 }
 
-func _isRegistered() (bool, error) {
-	services, err := filepath.Glob("/etc/zypp/services.d/*.service")
-	if err != nil {
-		fmt.Fprintln(os.Stderr, err)
-		return false, err
+func _getSUSEConnectStatus() (bool, bool, error) {
+	commandOutput, error := RunShellCommand(0, "SUSEConnect", "-s")
+	if error != nil {
+		fmt.Println(error)
+		return false, false, error
+	}
+	var suseConnectStatus []map[string]interface{}
+	json.Unmarshal([]byte(commandOutput), &suseConnectStatus)
+	subscriptionStatus := fmt.Sprintf("%v", suseConnectStatus[0]["subscription_status"])
+	status := fmt.Sprintf("%v", suseConnectStatus[0]["status"])
+	if strings.ToLower(status) == "registered" && strings.ToLower(subscriptionStatus) == "active" {
+		// system is registered with an active subscription
+		// check if services are present
+		services, err := filepath.Glob("/etc/zypp/services.d/*.service")
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return true, true, err
+		}
+		if len(services) == 0 {
+			errorMessage := "There are no services. Please, re register the system"
+			err := errors.New(errorMessage)
+			fmt.Println(errorMessage)
+			return true, true, err
+		}
+		// check if repos are present
+		repos, err := filepath.Glob("/etc/zypp/repos.d/*.repo")
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return true, true, err
+		}
+		if len(repos) == 0 {
+			errorMessage := "There are no repositores. Please, re register the system"
```

Comment:
This is also not correct, if there is a service, which we have if the code got this far, then the repo definitions do not matter as a `zypper ref -f` will restore the repos. That's the whole point of the service ;) Repository Information Service.

Reply:
Done, used he verbose option of `refs` option instead of `ref`